### PR TITLE
Filter listing by tag slug name or uuid on graphql query

### PR DESCRIPTION
**Pull request by joaodubas**
Add the option to search listings by list of tag slugs name or uuids. Example:

```graphql
query {
  listings(filters: {tagsSlug: ["salao-de-festas"]}) {
    listings {
      id
      address {
        street
        streetNumber
      }
    }
  }
}
```

### CODE REVIEW

File: apps/re_web/test/graphql/listings/query_test.exs [hunk 1]
```diff
@@ -305,6 +305,9 @@ defmodule ReWeb.GraphQL.Listings.QueryTest do
     end
 
     test "should query listing index with filtering", %{user_conn: conn} do
+      tag_1 = insert(:tag, name: "Tag 1", name_slug: "tag-1")
+      tag_2 = insert(:tag, name: "Tag 2", name_slug: "tag-2")
```
Comment (rhnonose) [inline apps/re_web/test/graphql/listings/query_test.exs]: I think there's a way of inserting the relationship without calling `Listings.upsert_tags/2`on every listing, maybe you should check `ex_machina`'s documentation.
If `tags: [build(:tag), build(:tag)]` directly doesn't work, you can create a `tag_listing_factory`. It should shrink the test a little bit.

File: apps/re_web/lib/graphql/types/listing.ex [hunk 2]
```diff
@@ -203,6 +203,8 @@ defmodule ReWeb.Types.Listing do
     field :garage_types, list_of(:garage_type)
     field :cities, list_of(:string)
     field :cities_slug, list_of(:string)
+    field :tags_slug, list_of(:string)
+    field :tags_uuid, list_of(:uuid)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/types/listing.ex]: These should be `list_of(non_null(:string))` and `list_of(non_null(:uuid))`.
The field can be `nil` but the elements shouldn't.

File: apps/re/test/listings/queries_test.exs [hunk 3]
```diff
@@ -60,4 +60,9 @@ defmodule Re.Listings.QueriesTest do
       assert expected == result
     end
   end
+
+  describe "remaining_count/2" do
+    test "get remaining count with tags" do
+    end
+  end
```
Comment (rhnonose) [inline apps/re/test/listings/queries_test.exs]: Did forget this hanging?

Comment (joaodubas) [reply apps/re/test/listings/queries_test.exs]: Ops, my bad! End up testing this through `Re.Listings.paginated`.

File: apps/re_web/lib/graphql/types/listing.ex (hunk 2 above)
Comment (joaodubas) [reply apps/re_web/lib/graphql/types/listing.ex]: Should I change the others attributes that are set to `list_of`?

Comment (rhnonose) [reply apps/re_web/lib/graphql/types/listing.ex]: If it's new and the clients didn't implement yet, yes.
Otherwise, it might break some queries that have specified the types in details.

Comment (joaodubas) [reply apps/re_web/lib/graphql/types/listing.ex]: 🆗

File: apps/re_web/test/graphql/listings/query_test.exs (hunk 1 above)
Comment (joaodubas) [reply apps/re_web/test/graphql/listings/query_test.exs]: You were right, rewrote some of the tests to add tags directly into listing, using the `insert` method.

File: apps/re/test/listings/queries_test.exs [hunk 4]
```diff
@@ -60,4 +62,35 @@ defmodule Re.Listings.QueriesTest do
       assert expected == result
     end
   end
+
+  describe "remaining_count/2" do
```
Comment (pmargreff) [inline apps/re/test/listings/queries_test.exs]: These tests look touching too much. If they should test only `remaining_count` they don't need to touch attributes or other functions as filters. A simple case is to create some listings and test the remaining count. So we don't touch `Filtering.apply` neither tags. 

Otherwise, if the intention was to create these integration test, maybe the describe message isn't appropriated. Does this comment make sense?

Comment (joaodubas) [reply apps/re/test/listings/queries_test.exs]: They're a testing an edge case for the `Re.Listings.Queries.remaining_count/2`.

I'll improve the description for both tests.

Comment (joaodubas) [reply apps/re/test/listings/queries_test.exs]: This is an example of what this test ensure: http://sqlfiddle.com/#!17/50fec/3
